b2sdk/v1/bucket.py

```diff
@@ -8,14 +8,18 @@
 #
 ######################################################################
 
+from .download_dest import AbstractDownloadDestination
+from .file_metadata import FileMetadata
 from .file_version import FileVersionInfoFactory
-from typing import Optional
+from typing import Optional, overload, Tuple
 from b2sdk import _v2 as v2
 from b2sdk.utils import validate_b2_file_name
 
 
 # Overridden to retain the obsolete copy_file and start_large_file methods
-# and to return old style FILE_VERSION_FACTORY attribute
+# and to retain old style FILE_VERSION_FACTORY attribute
+# and to retain old style download_file_by_name signature
```

Comment:
why we support `DonwloadedFile` in `download_file_by_id` but not in `download_file_by_name`?

Reply:
The only reason I added support for both new and old interface for `v1.Bucket` is that it is used internally (in Sync, i think), in code that is not apiver wrapped. Therefore it was necessary to overload it, so that old-style buckets would retain old-style interface for users but also could use the new interface when called by the SDK internally. There was no such need for `download_file_by_name` and since backporting features to old apiver v* is "best-effort", I didn't include it.